### PR TITLE
#4375 fix ios user redirect to set up password + update staging url for app association on native platforms

### PR DESCRIPTION
**Pull request by dklymenk**
### Details

This PR updates the list of associated domains with iOS app to make it identical with the one in AndroidManifest.xml
Also updates the staging url used for app association on both iOS and android.

### Fixed Issues

$ #4375 

### Tests / QA Steps

prod url test for iOS only:
1. Logout
2. Input an email address for new account
3. Check the inbox, tap the link in the email
4. If the "Choose app to open with" modal appears, select "Expensify"
5. An app should open 
6. Input the password
7. You should be automatically logged in

staging url test (ios and android):
1. Logout
2. Input an email address for new account
3. Check the inbox, long tap the link in the email. Tap "Copy" button.
4. Paste the link into any app that renders hyperlinks as clickable objects(I used calendar on ios and messages on android). Add `staging` to the URL, so it says `staging.new.expenisify.com...`.
5. Tap the link.
6. If the "Choose app to open with" modal appears, select "Expensify"
7. An app should open 
8. Input the password
9. You should be automatically logged in

### Tested On
- [x] iOS
- [x] andorid

#### iOS

prod url:

https://user-images.githubusercontent.com/64093836/128330262-7441caa8-ea29-48ee-86a6-abf4a360a2b9.mp4

staging url:

https://user-images.githubusercontent.com/64093836/128626813-fd3a4f38-e8b8-41e0-b4f2-fe53fc14f768.mp4

#### Android

https://user-images.githubusercontent.com/64093836/128628128-9313e05f-b63f-4845-81cc-334bfecf9723.mp4

### CODE REVIEW

File: ios/ExpensifyCash/Chat.entitlements [hunk 1]
```diff
@@ -6,9 +6,8 @@
 	<string>development</string>
 	<key>com.apple.developer.associated-domains</key>
 	<array>
-		<string>applinks:www.expensify.cash</string>
 		<string>applinks:staging.expensify.cash</string>
```
Comment (AndrewGable) [inline ios/ExpensifyCash/Chat.entitlements:9]:
```suggestion
		<string>applinks:staging.new.expensify.com</string>
```

**Review by AndrewGable — CHANGES_REQUESTED** [review]

File: ios/ExpensifyCash/Chat.entitlements (hunk 1 above)
Comment (AndrewGable) [reply ios/ExpensifyCash/Chat.entitlements:9]: Can you make this change on android too? Thanks: https://github.com/Expensify/App/blob/main/android/app/src/main/AndroidManifest.xml#L58

Comment (dklymenk) [reply ios/ExpensifyCash/Chat.entitlements]: Hello, the requested changes have been implemented.

**Comment by AndrewGable** [comment]
Code looks great, can you add a note on how to test the latest commit e203903? Thanks.

**Review by AndrewGable — APPROVED** [review]

**Comment by dklymenk** [comment]
> Code looks great, can you add a note on how to test the latest commit [e203903](https://github.com/Expensify/App/commit/e2039031d753f84142b26831f780f9e956de57ed)? Thanks.

Hello, I don't think you can test it. The magic link in emails is always for new.expensify.com, don't think you can generate one specifically for staging.

Or do you mean to test app association with any link, not magic link specifically.

**Comment by AndrewGable** [comment]
You can ask the tester to copy the `new.expensify.com` and paste into a note app. Then replace the `new.expensify.com` link with `staging.new.expensify.com` in a note app, then click the link. We do send staging links from our staging server, but this might be easiest for you and QA.

**Comment by dklymenk** [comment]
> You can ask the tester to copy the `new.expensify.com` and paste into a note app. Then replace the `new.expensify.com` link with `staging.new.expensify.com` in a note app, then click the link. We do send staging links from our staging server, but this might be easiest for you and QA.

OK, I will add that later. Not on my laptop right now.

**Comment by dklymenk** [comment]
@AndrewGable Hello, I have updated the original post with additional videos and instructions. Also the title is updated.

**Review by AndrewGable — APPROVED** [review]
Looks great, thank you!

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-08-24.  :confetti_ball:

**Pull request merged**